Kernel/Process.cpp

```diff
@@ -1908,8 +1908,14 @@ int Process::sys$open(const Syscall::SC_open_params* user_params)
     if (result.is_error())
         return result.error();
     auto description = result.value();
-    description->set_rw_mode(options);
-    description->set_file_flags(options);
+    if (description->file_flags()) {
+        // We already have file flags set on this description, so
+        // it must be a preopen description (probably, /proc/pid/fd).
+        // So don't reset its flags and r/w mode.
```

Comment:
Hmm, are we sure there isn't a sneaky way to turn a read-only FD into a writable one here?

Reply:
I've thought about this, and no. We're specifically giving out the same fd as existed previously, we're not changing its mode at all, so if it was read-only it'll stay read-only. Moreover, ProcFS won't even let you open it for writing in that case, but even if it would, that writability would ge ignored here and you'd get the same read-only fd.

Reply:
OK.

Reply:
Oh, except comparing `file_flags` to zero is not enough to check they're actually not set :(

That would have to be fixed separately now.